EIPS/eip-beacon_chain_push_withdrawals.md

```diff
@@ -0,0 +1,130 @@
+
+This transaction type has no associated gas costs.
+
+Implementations should take care to decode the correct amount of wei from the `EncodedWei` value, which represents a 256-bit unsigned integer value in little-endian byte order.
+
+TODO: add logs?
+
+## Rationale
+
+### Push vs pull approach
+
+This push approach gives validators a small subsidy with respect to processing, in lieu of needing to buy gas via normal EVM processing that would be required for a pull-based approach.
+
+This style also happens automatically when the requisite conditions are met on the beacon chain which is nicer UX for validators.
+
+### Why a new transaction type?
+
+This EIP suggests a new transaction type as it has special semantics different from other existing types of EVM transactions.
+
+An entirely new transaction type firewalls off generic EVM excution from this type of processing to simplify testing and security review of withdrawals.
+
+### Why no (gas) costs for new transaction type?
+
+The maximum number of this transaction type that can reach the execution layer at a given time is bounded (enforced by the consensus layer) and this limit is kept small so that
+any execution layer operational costs are negligible in the context of the broader block execution.
+
+### Why only balance updates? No general EVM execution?
+
+More general processing introduces the risk of failures, which complicates accounting on the beacon chain.
+
+This EIP suggests a route for withdrawals that provides most of the benefits for a minimum of the (complexity) cost.
+
+### Why new block validations?
+
+The beacon chain must be able to efficiently validate that the withdrawal transactions in a given execution block are
+the ones expected based on its own internal scheduling logic to maintain the soundness of the withdrawal mechanism.
+
+By requiring all withdrawal transactions to be at the front of every block where they are applicable, the algorithm to
+check consistency becomes a straightforward linear walk from the start of the list until a known, bounded (small) number.
+
+Having a simple ordering scheme like this facilitates optimizations clients may do with respect to withdrawal processing, which
+would be hampered if withdrawal transactions could be placed in the block freely.
+
+### SSZ vs RLP, and their boundary
+
+The beacon chain produces data with the SSZ serialization scheme. The execution layer currently uses RLP.
+
+There has to be some translation boundary from SSZ to RLP here and this EIP tries to minimize the number of
+places new logic is needed in either location.
+
+## Backwards Compatibility
+
+No issues.
+
+## Reference Implementation
+
+A draft PR containing a prototype implementation in Geth can be found here:
+
+https://github.com/ethereum/go-ethereum/pull/24468
+
+## Security Considerations
+
+Consensus-layer validation of withdrawal transactions is critical to ensure that the proper amount of ETH is withdrawn back into the execution layer.
```

Comment:
Additional security concern: this type of transaction may not be propagated through anything else than the engine API or trusted blocks (i.e. part of a chain recognized by the consensus layer). The tx-pool could be spammed with useless non-processable withdrawal transactions, and execution payload builders shouldn't consider these during building, or the payload will be invalid.

Reply:
good call! 

It might be worth defining this more broadly as a "system TX" or a "CL TX" and none of those types of TXs can be gossiped